test/test_puma_server.rb

```diff
@@ -152,6 +152,20 @@ def test_streaming_body
     assert_equal "Hello World", data.split("\r\n\r\n", 2).last
   end
 
+  def test_request_logs_with_custom_logger
+    conf = Puma::Configuration.new { |c| c.load 'test/config/custom_logger.rb' }
+    conf.load
+
+    server_run(custom_logger: conf.options[:custom_logger], log_requests: true) do |env|
+      [200, {}, [env["SERVER_NAME"], "\n", env["SERVER_PORT"]]]
+    end
+
+    data = send_http_and_read "GET / HTTP/1.0\r\nHost: 123.123.123.123:456\r\n\r\n"
+    assert_match(%r!Custom logging: 127\.0\.0\.1.*GET / HTTP/1\.0.*Host: 123\.123\.123\.123:456!, data)
+    data = send_http_and_read "GET / HTTP/1.1\r\nHost: 123.123.123.123\r\n\r\n"
+    assert_match(%r!Custom logging: 127\.0\.0\.1.*GET / HTTP/1\.1.*Host: 123\.123\.123\.123!, data)
```

Comment:
`send_http_and_read` reads the HTTP response, so you wont see the request logging there

I think the test in `test/test_integration_single.rb` should be sufficient